crates/swc_ecma_transforms_module/src/system_js.rs

```diff
@@ -0,0 +1,1134 @@
+                        span: DUMMY_SP,
+                        body: Some(BlockStmt {
+                            span: DUMMY_SP,
+                            stmts: self.build_module_item_export_all(meta),
+                        }),
+                        is_generator: false,
+                        is_async: false,
+                        type_params: Default::default(),
+                        return_type: Default::default(),
+                    },
+                }
+                .as_arg(),
+            ));
+        }
+
+        let execute = Function {
+            params: vec![],
+            decorators: Default::default(),
+            span: DUMMY_SP,
+            body: Some(BlockStmt {
+                span: DUMMY_SP,
+                stmts: execute_stmts,
+            }),
+            is_generator: false,
+            is_async: self.tla,
+            type_params: Default::default(),
+            return_type: Default::default(),
+        };
+
+        let return_stmt = ReturnStmt {
+            span: DUMMY_SP,
+            arg: Some(Box::new(Expr::Object(ObjectLit {
+                span: DUMMY_SP,
+                props: vec![
+                    PropOrSpread::Prop(Box::new(Prop::KeyValue(KeyValueProp {
+                        key: quote_ident!("setters").into(),
+                        value: Box::new(Expr::Array(setters)),
+                    }))),
+                    PropOrSpread::Prop(Box::new(Prop::KeyValue(KeyValueProp {
+                        key: quote_ident!("execute").into(),
+                        value: Box::new(Expr::Fn(FnExpr {
+                            ident: None,
+                            function: execute,
+                        })),
+                    }))),
+                ],
+            }))),
+        };
+
+        let mut function_stmts = vec![use_strict()];
+
+        if !self.declare_var_idents.is_empty() {
+            function_stmts.push(Stmt::Decl(Decl::Var(VarDecl {
+                span: DUMMY_SP,
+                kind: VarDeclKind::Var,
+                declare: false,
+                decls: self
+                    .declare_var_idents
+                    .iter()
+                    .map(|i| VarDeclarator {
+                        span: i.span,
+                        name: Pat::Ident(BindingIdent {
+                            id: i.clone(),
+                            type_ann: None,
+                        }),
+                        init: None,
+                        definite: false,
+                    })
+                    .collect(),
+            })));
+        }
+        function_stmts.append(&mut before_body_stmts);
+        function_stmts.push(return_stmt.into());
+
+        let function = Function {
+            params: vec![
+                Param {
+                    span: DUMMY_SP,
+                    decorators: Default::default(),
+                    pat: quote_ident!("_export").into(),
```

Comment:
Also, is `quote_ident` correct? Wnat should happen if there's a variable named `_export`? I don't know well about system js

Reply:
Systemjs use `_export` let one module can export somethings. Eg. `export const a = 1` will be transformed to  `export('a', 1)`. You can read docs from here  https://github.com/systemjs/systemjs/blob/main/docs/system-register.md#format-definition.